### ISSUE TITLE
Allow GET webhooks

### ISSUE THREAD

**Issue by geromueller** [gogits/gogs#2598]
Jenkins uses simple GETs to trigger builds. It would be great to be able to use the variables from the json file to format the url.

**Comment by Unknwon** [comment]
Thanks your feedback!

> It would be great to be able to use the variables from the json file to format the url.

Can you rephrase this sentence?

**Comment by geromueller** [comment]
You can add &cause=reason to the url, which is then shown for the build. I would like to format the url like:

https://myjenkins.org/job/mybuild/build?token=abc&cause=Pushed+by+{{pusher.name}}

**Comment by Unknwon** [comment]
Thanks, so what does this to do with GET webhooks in the title?

**Comment by geromueller** [comment]
The main request ist: add the option to use GET instead of POST.
But then there is no JSON to transmit. Insetad, add the possibility to use the variables from the json file, in building the GET URL.
Thanks!

**Comment by Unknwon** [comment]
Thanks, got it now!

But this feature request may have low priority, and involves a design problem of how to make a way to build the GET URL.

**Comment by geromueller** [comment]
I guess text/template would be the natural choice.

**Comment by geromueller** [comment]
Hi! Just to let you know: Jenkins also works with POST requests. It was an authorization problem before. I suggest you make this request very low priority, or even drop it.

**Comment by Unknwon** [comment]
:sweat_smile: Cool, so I'll reopen if someone else requests for it.

**Comment by antiloop222** [comment]
Hello, just to let you know that a POST request to Jenkins needs a crumb header to be passed, which is not requested for GET. So IMHO, it could be nice to have a GOGS webhook that only send a GET instead of a POST with a payload that is not necessary for Jenkins by the way

**Comment by Unknwon** [comment]
@antiloop222 thanks for the info! But how do you send the whole body via GET? Not sure what is the correct way.

**Comment by antiloop222** [comment]
Hello, I do not send anything with GET and it is enough for Jenkins to trig the push : in other words, the payload built by Gogs is not useful for Jenkins

**Comment by Unknwon** [comment]
OK, make sense.
